### ISSUE TITLE
definition of "rtlsdr_set_sample_freq_correction_f"

### ISSUE THREAD

**Issue by shuwang1** [mlaaks/coherent-rtlsdr#1]
Hello mlaaks,

A great job firstly! thumb up!

May I ask,  what's the difference between "rtlsdr_set_sample_freq_correction_f"  and "rtlsdr_set_sample_freq_correction"?  Sorry, I couldn't find the definition of "rtlsdr_set_sample_freq_correction_f"

Should 

ppm = (int) ( f / fcenter * 1.0e6 )

Cheers.

Shu

**Comment by mlaaks** [comment]
Hello,

rtlsdr_set_sample_freq_correction is the default function from librtlsdr, which setst the CFO correction in PPM units. It controls the resampler. The '_f' version is from Keenerds experimental fork of librtlsdr (again forked by tejeez, from whom I inherited this project). This version simply removes that PPM 1e6 scaling from the resampler control. A modified version of the librtlsdr is required anyhow, since we need to switch off tuner frequency synthesizer PLL dithering. With the dithering enabled. the relative phases of the channels will drift too fast fast.

All this is because we (ab)use the RTL2832U built-in resampler to time-align the signal channels. Otherwise, I would have needed to implement a ringbuffering scheme which would consume cycles and furthermore, get it working on the 10th attempt... I did consider this during my Master's, just to compare the CPU consumption between the implementations.

I wish I had more time to devote to this project some day, but I'm currently working on a PhD.

Tejeez librtlsdr for can be found at: 

https://github.com/tejeez/rtl-sdr

-Mikko
